### PR TITLE
Increase fluentd rolling-upgrade maxUnavailable to large value

### PR DESCRIPTION
**Pull request by shyamjvs**
~For testing wrt https://github.com/kubernetes/kubernetes/issues/61190#issuecomment-374899752~
Fixes issue https://github.com/kubernetes/kubernetes/issues/61190 wrt slow rolling-upgrade

/cc @x13n @wojtek-t 
/sig instrumentation
/kind bug
/priority critical-urgent

```release-note
NONE
```

### CODE REVIEW

File: cluster/addons/fluentd-gcp/scaler-deployment.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
   namespace: kube-system
   labels:
     k8s-app: fluentd-gcp-scaler
-    version: v0.3.0
+    version: v0.2.0
```
Comment (x13n) [inline cluster/addons/fluentd-gcp/scaler-deployment.yaml]: This just uses older version, how about disabling this addon altogether?

Comment (shyamjvs) [reply cluster/addons/fluentd-gcp/scaler-deployment.yaml]: Sorry.. changed at the wrong place. Should've changed `--ds-name=...`. Done.

**Comment by shyamjvs** [comment]
From the above run, the following is the 99% mem-usage of controller-manager with the scaler disabled:

```
   {
      "Name": "kube-controller-manager-e2e-10-master/kube-controller-manager",
      "Cpu": 0.12105515,
      "Mem": 300953600
    },
```

Next I'll turn it on and see (with some extra verbose c-m logs).

**Comment by shyamjvs** [comment]
So after turning it back on, here's goes up the mem-usage:

```
    {
      "Name": "kube-controller-manager-e2e-11-master/kube-controller-manager",
      "Cpu": 0.163745444,
      "Mem": 375926784
    },
```

Next I'll take a look at the ds-controller logs to see what's changing in the ds object.

**Comment by x13n** [comment]
FWIW, you can set SLEEP_SECONDS to some big number to ensure the scaler does only one run and sleeps forever. This will mean a few `kubectl get` calls and a single `kubectl set resources`.

**Comment by shyamjvs** [comment]
I've posted my findings from the experiments above here - https://github.com/kubernetes/kubernetes/issues/61190#issuecomment-375267584

**Comment by shyamjvs** [comment]
I'm done with my experiments here and updated the PR with the fix I explained in https://github.com/kubernetes/kubernetes/issues/61190#issuecomment-375269467.
@x13n @wojtek-t - PTAL.

**Comment by shyamjvs** [comment]
@x13n I've added no release-note. Let me know if you think I should.
@wojtek-t Could you approve?

**Comment by x13n** [comment]
@shyamjvs I think it's fine without one.

**Comment by wojtek-t** [comment]
> @wojtek-t Could you approve

Sorry - didn't see that PR and discussion before.
I'm not 100% sure convinced that this is actually a good change. The fact that something fixes tests doesn't necessary mean this is a good change.

While I agree with the fact that in this case applications on different nodes are independent (so it's not about application downtime), this is not the only thing we should keep in mind.
In majority of cases, the reason behind rolling-upgrade is actually canary-ing  the new version.

With this change there is a potential to break everything at once.
You may claim that upgrade of resources isn't particularly risky, but:

1. this strategy is not only for update of resources - the same strategy will be used for upgrading version (which obviously is potentially risky)

2. even updating resources for all pods at once may cause signifcant problems for users applications (which may be preempted due to that) - if there weren't enough resources in the cluster, this may cause significant problems. With rolling upgrade, admins may observe some preemptions and react - with all-at-once approach, suddenly many things may get preempted.

So I would be very careful with that change.
And I would at least like to hear other opinions about that change. Adding some folks:
@kow3ns @kubernetes/sig-apps-misc 
@roberthbailey @kubernetes/sig-cluster-lifecycle-misc

**Comment by shyamjvs** [comment]
Those are good points about risky upgrades and potential evictions - and I agree we need more opinion here. Adding a couple of my thoughts:
- By having lower maxUnavailable, we may be trading off responsiveness (i.e fluentds not coping up with "log load" quickly enough) for safety (i.e newer version potentially crashing). I don't have much to say which one is more important (for e.g if cluster admin is fairly confident about newer version, he/she may want the rollout quicker)
- Wrt evictions - From what I remember talking to @x13n (please confirm), the scalingPolicy is intended to be more like a manual knob for cluster-admin to change resources. So here again we may either say "the admin should understand eviction effect of his change before doing it" or "assume the admin can make blunders unknowingly". I would be more inclined towards the latter, but again I'm no cluster-ops expert.

**Comment by x13n** [comment]
Yes - right now this is to be a manual knob to be used by the cluster admin. In the future, however, some form of (CP?)VPA should be used instead and the admin will be able to just tune the parameters. How does VPA solve this problem? Is it just operating on Pod, not DS or Deployment level?
cc @kubernetes/sig-autoscaling-misc

**Comment by jdumars** [comment]
@crassirostris @piosz @x13n PTAL for approval ASAP so this can make 1.10 on Monday

**Comment by shyamjvs** [comment]
Removing this from 1.10 milestone as we do not want this change to go into it. For master branch, we'll create a revert of this.
Rationale explained here - https://github.com/kubernetes/kubernetes/issues/61190#issuecomment-376150569

**Pull request merged**